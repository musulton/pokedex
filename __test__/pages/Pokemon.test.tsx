import {render} from "@testing-library/react";

import Pokemon from "../../src/pages/pokemon";
import React, {useState} from "react";

jest
    .mock("../../src/components/ImageCard", () => "ImageCard")
    .mock("../../src/services/api", () => ({
        fetchData: jest.fn()
    }))
    .mock("../../src/utils/common", () => ({
        ...jest.requireActual("../../src/utils/common"),
        getRandomColor: jest.fn().mockImplementation(() => "white")
    }))
    .mock('react', () => ({
        ...jest.requireActual('react'),
        useState: jest.fn(),
    }))

describe("Pokemon pages", () => {
    const mockState = {
        count: 100,
        pokemon: [{
            id: 1,
            name: "Pokemon",
            imageSrc: "http://gugel.com"
        }]
    }

    beforeEach(() => {
        useState.mockImplementation(() => [mockState, jest.fn()])
    })

    it("should render Pokemon correctly", () => {
        const {container} = render(<Pokemon />)

        expect(container.firstChild).toMatchSnapshot();
    })
})
