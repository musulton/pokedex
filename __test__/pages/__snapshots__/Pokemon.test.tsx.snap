// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon pages should render Pokemon correctly 1`] = `
<div
  class="sc-249c5c88-0 bHOtmB"
>
  <p
    class="sc-249c5c88-1 dUHQTv"
  >
    Loading...
  </p>
</div>
`;
